components/CoverletterHeader.vue

```diff
@@ -0,0 +1,62 @@
+<template>
+  <header>
```

Comment:
since your validator is checking that every prop must be existing, you should rather put a single v-if at the <header> element to check that the address object is there. `<header v-if="address" ...`